app/src/behaviors/behavior_combo.c

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_combo
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <sys/dlist.h>
+#include <kernel.h>
+
+#include <zmk/behavior.h>
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct combo_t {
+    int32_t key_positions[CONFIG_ZMK_BHV_COMBO_MAX_KEYS_PER_COMBO];
+    int32_t key_position_len;
```

Comment:
Instead, can we fill the rest with MAX to avoid extra storage per combo?

Reply:
This prevents looping over the items to count in other places, so I think it's worth it.

Reply:
`combo_cfg`